### ISSUE TITLE
infra: improve release workflow

### ISSUE THREAD

**Issue by sondove** [novem-code/novem-python#51]
It seems a bit excessive to merge in a release bump, it would be nice if all the steps to cut a release could be done automatically as part of the github release workflow.

Basically run `make bump version` into a direct commit and then tag that.

**Comment by myme** [comment]
I think making use of GitHub's release workflow would be nice. With a GitHub workflow that triggers on release and does all the things.

![image](https://github.com/user-attachments/assets/6c493de2-0235-4355-a701-8c016bc43ddf)

**Comment by sondove** [comment]
> I think making use of GitHub's release workflow would be nice.

Yeah, I believe we have almost everything, except bumping the version as part of the release workflow. (Which i guess would require making a commit and tagging that new commit)
